modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1531,6 +1531,53 @@ private void _fillBusinessTypePicklistDefaultValue(
 		}
 	}
 
+	private Predicate _fillCustomFieldPredicate(
+		Column<?, Object> column, ObjectField objectField, String search) {
+
+		Predicate customFieldPredicate = null;
+		String businessType = _getBusinessType(objectField);
+
+		List<String> textInstances = new ArrayList<>();
+
+		textInstances.add(ObjectFieldConstants.BUSINESS_TYPE_TEXT);
+		textInstances.add(ObjectFieldConstants.BUSINESS_TYPE_LONG_TEXT);
+		textInstances.add(ObjectFieldConstants.BUSINESS_TYPE_RICH_TEXT);
+		textInstances.add(ObjectFieldConstants.BUSINESS_TYPE_TEXT);
+
+		List<String> numberInstances = new ArrayList<>();
+
+		numberInstances.add(ObjectFieldConstants.BUSINESS_TYPE_INTEGER);
+		numberInstances.add(ObjectFieldConstants.BUSINESS_TYPE_LONG_INTEGER);
+
+		List<String> doubleInstances = new ArrayList<>();
+
+		doubleInstances.add(ObjectFieldConstants.BUSINESS_TYPE_DECIMAL);
+		doubleInstances.add(ObjectFieldConstants.DB_TYPE_BIG_DECIMAL);
+		doubleInstances.add(
+			ObjectFieldConstants.BUSINESS_TYPE_PRECISION_DECIMAL);
+
+		if (textInstances.contains(businessType)) {
```

Comment:
I was thinking another solution instead of creating a List. The Object Field contains the database type, the method is objectField.getDBType(). You could refactor this condition to compare if the DBType is String or Clob;

number instances -> integer/long
double instances -> bigdecimal/double
(there are constants for the DB_TYPE in ObjectFieldConstants)

So, we wouldn't need to create any list. Just refactoring the conditions. What do you think?

Reply:
I updated!